components/mosart/src/riverroute/MOSART_RES_type.F90

```diff
@@ -0,0 +1,321 @@
+     real(r8), pointer :: wres_t(:,:)    ! MOSART reservoir storage (m3 for water, kg for mud and sand sediment)
+     real(r8), pointer :: dwres_t(:,:)   ! MOSART reservoir storage change (m3 for water, kg for mud and sand sediment)
+     !! exchange fluxes
+     real(r8), pointer :: eres_in_t(:,:) ! MOSART reservoir inflow  (m3/s for water, kg/s for mud and sand sediment) 
+     real(r8), pointer :: eres_out_t(:,:)! MOSART reservoir outflow  (m3/s for water, kg/s for mud and sand sediment) 
+
+    end type TstatusFlux_reservoir
+
+    type (Treservoir_control)   ,   public :: Tres_ctl
+    type (Tpara_reservoir),   public :: Tres_para
+    type (TstatusFlux_reservoir),   public :: Tres
+
+  public :: MOSART_reservoir_init
+
+!-----------------------------------------------------------------------
+  contains
+!-----------------------------------------------------------------------
+
+  subroutine MOSART_reservoir_init
+! !DESCRIPTION:
+! initialize MOSART-reservoir variables
+! 
+! !USES:
+! !ARGUMENTS:
+!
+! !REVISION HISTORY:
+! Author: Hongyi Li
+!
+
+     ! !DESCRIPTION: initilization of reservoir module
+     implicit none
+
+    integer :: ier                  ! error code
+    integer :: begr, endr, iunit, nn, n, cnt, nr, nt
+    integer  :: damID
+    character(len=*),parameter :: subname = '(MOSART_reservoir_init)'
+    character(len=*),parameter :: FORMI = '(2A,2i10)'
+    character(len=*),parameter :: FORMR = '(2A,2g15.7)'
+    type(file_desc_t):: ncid       ! netcdf file
+    type(var_desc_t) :: vardesc    ! netCDF variable description
+    type(io_desc_t)    :: iodesc_dbl ! pio io desc
+
+    begr = rtmCTL%begr
+    endr = rtmCTL%endr  
+
+    if(endr >= begr) then
+        allocate(Tres_para%Tres(begr:endr))
+        Tres_para%Tres = 0._r8
+        allocate(Tres_para%Eff_trapping(begr:endr))
+        Tres_para%Eff_trapping = 0._r8
+        do iunit=begr,endr
+            damID = WRMUnit%INVicell(iunit) 
+            if (.not.(damID > ctlSubwWRM%LocalNumDam .OR. damID <= 0 .or. WRMUnit%MeanMthFlow(damID,13) <= 0.01_r8)) then
+                Tres_para%Tres(iunit) = CRTres(WRMUnit%StorCap(damID), WRMUnit%MeanMthFlow(damID,13))
+                Tres_para%Eff_trapping(iunit) = CREff_trapping(Tres_para%Tres(iunit))
+                !write(iulog,*) ' Reservoir Trapping ', iunit, WRMUnit%StorCap(damID), WRMUnit%MeanMthFlow(damID,13), Tres_para%Tres(iunit), Tres_para%Eff_trapping(iunit)
+            else
+                Tres_para%Tres(iunit) = 0._r8
+                Tres_para%Eff_trapping(iunit) = 0._r8
+            end if
+        end do    
+
+        allocate (Tres%wres(begr:endr,nt_rtm))
+        Tres%wres = 0._r8
+
+        allocate (Tres%dwres(begr:endr,nt_rtm))
+        Tres%dwres = 0._r8
+
+        allocate (Tres%eres_in(begr:endr,nt_rtm))
+        Tres%eres_in = 0._r8
+
+        allocate (Tres%eres_out(begr:endr,nt_rtm))
+        Tres%eres_out = 0._r8    
+
+if(0>1) then ! comment out temporily, please do not delete
```

Comment:
Could you correct the indentation of this code block?

Reply:
In addition, the initialization of Tres_para%Eff_trapping_t seems to have done twice.

Reply:
Done and fixed.